### PR TITLE
Set filtered jar's manifest time to epoch

### LINKED ISSUE #43297: Using jib with opentelemetry cannot produce a stable library layer
### Describe the bug

Using opentelemetrey extension with jib makes the first layer unstable as the opentelemetry exporter jar's content change between each build.

### Expected behavior

The jar of a dependency should not change between each build. Or it should be put in a separate layer.

### Actual behavior

_No response_

### How to Reproduce?

[https://github.com/Malandril/quarkus-otel-jib](https://github.com/Malandril/quarkus-otel-jib)

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

3.14.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

I think it is caused by the `RemovedResourceBuildItem` updating the date of the Manifest.md in the jars where the resources are removed.

This is a diff between two successive compilation:
```
> Archive:  build/quarkus-build/gen/quarkus-app/lib/main/io.opentelemetry.opentelemetry-exporter-otlp-1.39.0.jar
4c4
<       177  2024-09-14 22:22   META-INF/MANIFEST.MF
---
>       177  2024-09-14 22:26   META-INF/MANIFEST.MF
```
So i think its not only limited to the opentelemetry extension, but to any extension using a `RemovedResourceBuildItem`

### PR DESCRIPTION
**Pull request by Malandril**
Makes layer stable by setting setting the time of the Manifest to epoch when a jar was modified by a `RemovedResourceBuildItem`.

- Fixes: #43297

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java [hunk 1]
```diff
@@ -1302,10 +1307,7 @@ static void filterJarFile(Path resolvedDep, Path targetPath, Set<String> transfo
                             entry.setCompressedSize(-1);
                             out.putNextEntry(entry);
                             try (InputStream inStream = in.getInputStream(entry)) {
-                                int r = 0;
```
Comment (gastaldi) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java:1305]: This is intentional and we have tested before, reusing the buffer is faster in this case

**Review by gastaldi — CHANGES_REQUESTED** [review]

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java (hunk 1 above)
Comment (Malandril) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java]: Ok thanks

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java [hunk 2]
```diff
@@ -1290,7 +1291,12 @@ static void filterJarFile(Path resolvedDep, Path targetPath, Set<String> transfo
                 } else {
                     manifest = new Manifest();
                 }
-                try (JarOutputStream out = new JarOutputStream(Files.newOutputStream(targetPath), manifest)) {
+                try (JarOutputStream out = new JarOutputStream(Files.newOutputStream(targetPath))) {
+                    ZipEntry manifestEntry = new ZipEntry(JarFile.MANIFEST_NAME);
+                    // Set manifest time to epoch to always make the same jar
+                    manifestEntry.setTime(0);
```
Comment (gastaldi) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java:1297]: Why not call this before the try block and keep the manifest being passed as part of the `JarOuputStream constructor?

Comment (gastaldi) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java:1297]: Ignore this, I see it is set in the ZipEntry, not in the Manifest

**Comment by gastaldi** [comment]
Can you please add a test in `JarResultBuildStepTest`? Also, I think it's missing a call to `out.closeEntry()` after the writes (including the loop that iterates through each entry).

Lastly, since `JarEntry extends ZipEntry`, let's use that instead (to avoid another import) :)

**Review by gastaldi — APPROVED** [review]
Great job!

**Pull request merged**